crates/cheatnet/src/cheatcodes/l1_handler_execute.rs

```diff
@@ -1,114 +1,101 @@
-use crate::constants::build_block_context;
-use crate::CheatnetState;
-use anyhow::{anyhow, Context, Result};
-use blockifier::execution::{
-    errors::EntryPointExecutionError,
-    execution_utils::{felt_to_stark_felt, felts_as_str},
-};
-use blockifier::state::cached_state::CachedState;
-use blockifier::transaction::{
-    errors::TransactionExecutionError,
-    transactions::{ExecutableTransaction, L1HandlerTransaction},
-};
+use super::CheatcodeError;
+use crate::state::BlockifierState;
+use anyhow::Result;
 use cairo_felt::Felt252;
-use cairo_lang_starknet::contract::starknet_keccak;
-use starknet_api::core::{ContractAddress, EntryPointSelector};
-use starknet_api::hash::StarkFelt;
-use starknet_api::transaction::{Calldata, Fee, TransactionHash};
+use starknet_api::core::ContractAddress;
 
-use super::{CheatcodeError, EnhancedHintError};
-use crate::panic_data::try_extract_panic_data;
-use crate::state::ExtendedStateReader;
-
-impl CheatnetState {
+impl BlockifierState<'_> {
     pub fn l1_handler_execute(
         &mut self,
-        contract_address: ContractAddress,
-        function_name: &Felt252,
-        from_address: &Felt252,
-        paid_fee_on_l1: &Felt252,
-        payload: &[Felt252],
+        _contract_address: ContractAddress,
+        _function_name: &Felt252,
+        _from_address: &Felt252,
+        _paid_fee_on_l1: &Felt252,
+        _payload: &[Felt252],
     ) -> Result<(), CheatcodeError> {
-        let blockifier_state: &mut CachedState<ExtendedStateReader> = &mut self.blockifier_state;
+        panic!("l1_handler_execute is temporarily not supported in this version, use a different version of starknet foundry");
+        // TODO this will not work with the current design as transactions cannot be executed on State trait. We should make it work.
```

Comment:
We probably could downcast it but I am not sure if it is the desired behaviour (to execute transaction in the middle of other transaction) assuming that each test case is an transaction

Reply:
What about declare then? I think it should still be a transaction, even though it is not rly a desired behavior. I don't it changes much from the user perspective

Reply:
Why do you think it should be a transaction?

Reply:
It is a transaction on starknet, I don't think it is worth to do a workaround. We can just use blockifier's api this way, otherwise we have to do some magic here.

Reply:
>It is a transaction on starknet

Deploy is one as well, I don't see that as a reason.

>We can just use blockifier's api this way, otherwise we have to do some magic here.

We just need to use lower level blockifier API which adds contract class to the map, idk technically we can do this but I am not sure if this is good from the codebase consistency perspective.

Reply:
I guess using lower lvl api is fine then. When it comes to `deploy` - our `deploy` simulates `deploy_syscall` so we are consistent with starknet there

Reply:
cc @Arcticae